### ISSUE TITLE
Implement .map(items, callback, done)

### ISSUE THREAD

**Issue by tshemsedinov** [metarhia/MetaSync#16]
Use case:
```js
metasync.map(['Beijing', 'Roma', 'Kiev'], (item, callback) => {
  getCityPopulation(item, (count) => {
    callback(count);
  });
}, (err, arr) => {
  let total = arr.reduce((prev, cur) => (prev + cur), 0);
  console.log('Total population: ' + total);
});
```

**Comment by tshemsedinov** [comment]
Artur Mironenko, plz assign yourself to this issue

**Comment by burntheroad** [comment]
@tshemsedinov are there any issues with a current implementation? Looks like it works exactly like in the use case described here, except `callback(count)` misses `err` argument. 
Are there any failing tests?

**Comment by aqrln** [comment]
@burntheroad no, there aren't. This issue had been created before this function was implemented by me but hasn't been closed since I didn't notice it. You may fix #27 or any other issue instead. Or just write tests because there are quite a few of them in the library.
